tools/tuhi-live.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+
+import logging
+import os
+import pwd
+import sys
+import multiprocessing
+from multiprocessing import reduction
+
+manager = None
+logger = None
+
+
+def open_uhid_process(queue_in, conn_out):
+    while True:
+        try:
+            pid = queue_in.get()
+        except KeyboardInterrupt:
+            return 0
+        else:
+            fd = os.open('/dev/uhid', os.O_RDWR)
+            reduction.send_handle(conn_out, fd, pid)
```

Comment:
one example that I saw was using `fd.fileno()` here. maybe that makes a difference?

Reply:
It shouldn't. It depends if you are using `File` objects or if you are using, like here plain fds. The only difference is that with plain fds, you have to use `os.read/write` instead of `f.read/write`